docs/docs/aztec/developer/wallet-providers/account-abstraction.md

```diff
@@ -1,3 +1,113 @@
+
+## Accounts in Aztec
+
+Aztec has no concept of Externally Owned Accounts. Every account is implemented as a contract. Account contracts typically implement an `entrypoint` function that receives the actions to be carried out and an authentication payload. In pseudocode:
+
+```
+publicKey: PublicKey;
+
+def entryPoint(payload):
+    let { privateCalls, publicCalls, nonce, signature } = payload;
+    let payloadHash = hash(privateCalls, publicCalls, nonce);
+    validateSignature(this.publicKey, signature, payloadHash);
+        
+    foreach privateCall in privateCalls:
+        let { to, data, value } = privateCall;
+        call(to, data, value);
+        
+    foreach publicCall in publicCalls:
+        let { to, data, value, gasLimit } = publicCall;
+        enqueueCall(to, data, value, gasLimit);
+```
+
+Read more about how to write an account contract [here](./writing-an-account-contract.md).
+
+### Accounts and wallets
+
+Account contracts are tightly coupled to the wallet software that users use to interact with the protocol. Dapps submit to the wallet software an execution request (eg "call swap in X contract"), and the wallet encodes and authenticates the request as a valid payload for the user's account contract. The account contract then validates the request encoded by the wallet, and executes the actions requested by the dapp.
+
+### Execution requests
+
+Note that nothing related to signature verification or payload execution is enshrined in the protocol, since account contracts are free to define this entrypoint however they see fit. Therefore, a request for executing an action has a simpler structure than in Ethereum, and just requires:
+
+- The `origin` contract to execute as the first step
+- The initial function to call (usually `entrypoint`)
+- The arguments (which encode the private and public calls to run as well as any signatures)
+
+### Entrypoint restrictions
+
+Entrypoint methods are not enshrined in the protocol, and any function can be called as an entrypoint. The only restriction is that it must be private (not open), so all transactions are initiated with a client-side zero-knowledge proof.
+
+This means that, unlike other protocols, Aztec does not impose any restrictions on the actions that can be carried out during the validation phase of the entrypoint, since these actions are executed by the client and wrapped in a zero-knowledge proof that is verified by the network. You are free to call into other contracts, access storage, or do as much computing as you need in the entrypoint.
+
+### Nonces and replay protection
+
+Every execution in the protocol emits the hash of the request as a nullifier, preventing the same transaction from being executed more than once. Nonces, on the other hand, are left to the account contract implementation. This allows building accounts with strictly incremental nonces or where transactions can be processed out-of-order.
```

Comment:
> Every execution in the protocol emits the hash of the request as a nullifier...

It might be best to be more verbose here for clarity :thinking: Maybe "_Every_ transaction execution considered valid by the protocol emits the hash of the transaction execution request as a nullifier, ..."